### PR TITLE
Adds default.yml vs user.yml

### LINKED ISSUE #928: Update /users/creating-user-property
default.yml is just a UserProfileScchema with name “default”, it’s placed in user_profile_scchemas directory, in form you have to do

```
resource: User
fields:
  profiles:
    default: 
      properties:
        my_property:
```
 
and then in graphql you need to fetch values via profiles, whee’as user.yml is placed in app and it’s like adding property directly to user:

```
resource: User
fields:
  properties:
    my_property:
```
and then in graphql you fetch it also directly from user [ same as with models, no need to go through extra step of user profile.

At the end of the day, both approaches work and both have advantages and disadvantages. I find user.yml much easier to use, but it lacks namespace and you can’t add user property via module [ as this could generate tons of collisions ]

### PR DESCRIPTION
**Pull request by diana-lakatos**
Fixes #928

### CODE REVIEW

File: app/views/pages/developer-guide/users/creating-user-property.liquid [hunk 1]
```diff
@@ -31,6 +31,40 @@ properties:
     type: string
 ```
 
+#### Note on default.yml vs user.yml 
+
+**default.yml**
+
+Another approach could be to create `default.yml`: a UserProfileSchema with  the name “default” placed in user_profile_scchemas directory. In a form you should then do the following:
```
Comment (pavelloz) [inline app/views/pages/developer-guide/users/creating-user-property.liquid]:
```suggestion
Another approach could be to create `default.yml`: a UserProfileSchema with the name “default” placed in user_profile_schemas directory. In a form you should then do the following: 
```

File: app/views/pages/developer-guide/users/creating-user-property.liquid [hunk 2]
```diff
@@ -31,6 +31,40 @@ properties:
...
+fields:
+  properties:
+    my_property:
+```
+
+So in GraphQL you fetch it also directly from user (same as with models, no need to go through the extra step of using a user profile).
+
+**Conclusion**
+
+At the end of the day, both of these approaches work and both have advantages and disadvantages. `user.yml` is easier to use, but it lacks namespace and you can’t add user properties via module.
```
Comment (pavelloz) [inline app/views/pages/developer-guide/users/creating-user-property.liquid:66]: > it lacks namespace and you can’t add user properties via module. 

@Slashek expand on this

**Comment by diana-lakatos** [comment]
To be discussed: Should we remove mentions of user profiles? Many partners pointed out it pretty much the same as models. 

We mention default.yml in these topics:

- https://documentation.platformos.com/developer-guide/apis/persisting-part-of-third-party-api-response 
- https://documentation.platformos.com/developer-guide/platformos-workflow/codebase
- https://documentation.platformos.com/developer-guide/users/resetting-password-authenticated-user
- https://documentation.platformos.com/use-cases/e-commerce/configuring-data-schema

**Comment by diana-lakatos** [comment]
@Slashek Please advise what to do with this:

- explain user profiles (if so, we'll merge this PR) OR
- remove references to user profiles (if so, I'll open a ticket for that task)

Thank you!

**Comment by diana-lakatos** [comment]
@pavelloz I have discussed this with @Slashek, and as a first phase, we would like to remove references to `default.yml` everywhere in the documentation. 

I will open a separate ticket for that, and close this PR.

**Pull request closed without merging**